src/main/java/uk/gov/companieshouse/presenter/account/interceptor/UserAuthenticationInterceptor.java

```diff
@@ -0,0 +1,144 @@
+package uk.gov.companieshouse.presenter.account.interceptor;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import uk.gov.companieshouse.api.util.security.Permission.Key;
+import uk.gov.companieshouse.api.util.security.Permission.Value;
+import uk.gov.companieshouse.api.util.security.TokenPermissions;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.util.RequestLogger;
+
+import org.springframework.web.servlet.AsyncHandlerInterceptor;
+
+/**
+ * Checks each request to the service to verify that the user is authorised to
+ * use the service.
+ */
+@Component
+public class UserAuthenticationInterceptor implements AsyncHandlerInterceptor, RequestLogger {
+
+    private static final Key USER_PRESENTER = Key.USER_PRESENTER;
+    private static final String CREATE = Value.CREATE;
+
+    final Logger logger;
+    private final AuthenticationHelper authHelper;
+
+    /**
+     * Constructor which configures the logger and authentication helper class.
+     *
+     * @param authHelper the {@link AuthenticationHelper}
+     * @param logger     the specified logger
+     */
+    @Autowired
+    public UserAuthenticationInterceptor(final AuthenticationHelper authHelper, final Logger logger) {
+        this.authHelper = authHelper;
+        this.logger = logger;
+    }
+
+    /**
+     * Ensure requests are authenticated for a user.
+     */
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response,
+            Object handler) {
+
+        final String identityType = authHelper.getAuthorisedIdentityType(request);
+        boolean validIdentity = false;
+        boolean validPermission = false;
+        final String requestMethod = authHelper.getRequestMethod(request);
+        boolean isGetRequest = authHelper.isGetMethod(requestMethod);
+
+        if (requestMethod == null || requestMethod.isEmpty()) {
+            logger.debugRequest(request, "UserAuthenticationInterceptor error: no request method", null);
+        }
+
+        if (identityType == null || identityType.isEmpty()) {
+            logger.debugRequest(request, "UserAuthenticationInterceptor error: no authorised identity type", null);
```

Comment:
Anywhere something is incorrect like this, I think it would be cleaner to log an error and then return false. That way we know the requests getting to the bottom of the method are good and we don't need to recheck stuff.